src/Commands/HydroponicsDrushCommands.php

```diff
@@ -46,32 +46,64 @@ public function hydroponics(
     $active_queues = \Drupal::config('strawberryfield.hydroponics_settings')->get('queues');
     $done = [];
 
+    //track when queue are empty for n cycles
+    $idle = [];
 
     // Get which queues we should run:
 
     foreach($active_queues as $queue) {
-      $done[$queue] = $loop->addPeriodicTimer(1.0, function ($timer) use ($loop, $queue) {
+
+      // Set number of idle cycle to wait
+      $idle[$queue] = 3;
+
+//      $done[$queue] = $loop->addPeriodicTimer(1.0, function ($timer) use ($loop, $queue) {
+      $done[$queue] = $loop->addPeriodicTimer(1.0, function ($timer) use ($loop, $queue, &$idle) {
         error_log("Starting to process $queue");
+
         $number = \Drupal::getContainer()
           ->get('strawberryfield.hydroponics')
           ->processQueue($queue, 60);
         error_log("Finished processing $queue");
+
         if ($number == 0) {
           error_log("No items left for $queue");
-          $loop->cancelTimer($timer);
+
+          // decrement idle counter
+          $idle[$queue] -= 1;
+//          $loop->cancelTimer($timer);
+        }
+
+        else {
+          // no empty so reset idle counter
+          $idle[$queue] = 3;
         }
+
       });
     }
 
-    $loop->addTimer(720.0, function ($timer) use ($loop, $timer_ping, &$done) {
-      // Finish all if 360 seconds are reached
-      error_log("All Done, 720 Seconds past, clearing the timers");
-      $loop->cancelTimer($timer_ping);
-        foreach($done as $queue_timer) {
-          $loop->cancelTimer($queue_timer);
+//    $loop->addTimer(720.0, function ($timer) use ($loop, $timer_ping, &$done) {
+    $loop->addTimer(60.0, function ($timer) use ($loop, $timer_ping, &$done, &$idle) {
```

Comment:
@giancarlobi, this time here runs only once, at 60 seconds. I think what we need to make this happen is that this time runs every 60 seconds (periodical). and we keep the security timer around.
If the periodical finds that all queues are idle, then it stops the security (720 seconds) one and itself. If not it keeps running.
The security one stops everything even the new periodical one that runs every 60 seconds.
We can make these times a setting too?

Reply:
@DiegoPino You are right, my error, I was thinking in periodic but it doesn't flow from brain to keyborad ....

Reply:
> We can make these times a setting too?

I don't know if we really need make it a settings, what about wait and eventually add it in the future?